lib/client/src/client.rs

```diff
@@ -2459,6 +2487,30 @@ impl TransactionBuilder {
         Ok(signature)
     }
 
+    pub async fn send_and_confirm_with_sequence_check(
```

Comment:
I'm not happy with putting this function into TransactionBuilder
- currently it never needs to work with MangoClient or MangoAccountValue, it's an isolated generic utility
- adding the ix to the raw instruction list at this point means that it's cu budget isn't covered by the compute budget instruction the user may have added (typically via PreparedInstructions)

Imo it's better for users to add the sequence check to their PreparedInstructions